dvc/dependency/local.py

```diff
@@ -1,8 +1,53 @@
 from __future__ import unicode_literals
 
-from dvc.dependency.base import DependencyBase
+import os
+
+from dvc.pkg import Pkg
+
+from dvc.utils.compat import urlparse
+
+from dvc.dependency.base import DependencyBASE
 from dvc.output.local import OutputLOCAL
 
 
-class DependencyLOCAL(DependencyBase, OutputLOCAL):
-    pass
+class DependencyLOCAL(DependencyBASE, OutputLOCAL):
+    PARAM_PKG = "pkg"
+
+    def __init__(self, stage, info, remote=None):
+        pkg_info = info.pop(self.PARAM_PKG, None)
+
+        super(DependencyLOCAL, self).__init__(stage, info, remote=remote)
+
+        if not pkg_info:
+            return
+
+        self.pkg = Pkg(stage.repo.config, pkg_info)
+
+        if not self.pkg.installed:
+            self.pkg.install()
+
+        out_path = os.path.join(
+            self.pkg.repo.root_dir, urlparse(self.url).path.lstrip("/")
+        )
+
+        outs = self.pkg.repo.find_outs_by_path(out_path)
+        assert len(outs) == 1
+        out = outs[0]
+        self.path_info.path = out.cache_path
+
+        if not os.path.exists(out.cache_path):
+            self.pkg.repo.fetch(out.stage.path)
```

Comment:
also, don't quite clear why is Dependency doing the fetch

Reply:
Could you please help me to understand how the data file instatiation\checkout happens during the import?

Reply:
@shcheklein See https://github.com/iterative/dvc/pull/2012#discussion_r289763541

@dmpetrov is the question still relevant?